apps/system/js/utility_tray.js

```diff
@@ -236,7 +259,15 @@ window.UtilityTray = {
           evt.preventDefault();
         }
 
+        var oldDelta = this.lastDelta;
+
         this.onTouchMove(evt.touches[0]);
+
+        var move = this.lastDelta - oldDelta;
+        if (Math.abs(move) > 0) {
+          this.lastTime = evt.timeStamp;
+          this.lastMove = move;
+        }
```

Comment:
A couple of notes here. First, this logic would make more sense to me if it were inside the `onTouchMove` function. Relying on `onTouchMove' to modify `this.lastDelta` seems a little funky and would be clearer if it was inline in the function. I guess we would need to pass the event timeStamp into the `onTouchMove`.

Second, `this.lastMove` is the change in Y deltas between confirmed touched moves, which means it's giving us an acceleration/deceleration if I'm understanding it correctly. I guess the onTouchMove might not be called with constant dt, but assuming performance is relatively consistent, I think `this.lastMove` is more or less telling us the change in swipe velocity. Assuming that is the case, how does change in velocity vs. whether or not the utility tray is shown [1] tell us whether or not to use velocity based animation? Shouldn't we be more concerned about velocity of the swipe (ie whether that is negative or positive) rather than acceleration of the swipe?

1.) https://github.com/mozilla-b2g/gaia/pull/28842/files#diff-690ac4f812d69b66c808d3e430042345R432